cpp-client/build-dependencies.sh

```diff
@@ -11,6 +11,21 @@
 # Fail on first error; echo each command before executing.
 set -eux
 
+if [ "$#" -ge 1 -a "$1" == "--clean" ]; then
```

Comment:
We need to figure out how to manage this file, since it's now part of deephaven-base-images.

Reply:
You win, I am removing it.